cert/cert.go

```diff
@@ -120,25 +130,51 @@ func (ca *CA) Refresh() (bool, error) {
 		return false, err
 	}
 
-	if bytes.Equal(cert, ca.pem) {
+	isCACertSame, err := CompareCertificates(cert, ca.pem)
+	if err != nil {
+		log.Infof("cert: error comparing CA certificates")
+		return false, err
+	}
+
+	if isCACertSame {
 		if ca.File != nil {
 			log.Infof("cert: existing CA certificate at %s is current", ca.File.Path)
 		}
 		return false, nil
 	}
 
+	// If CA cert has changed, write out new CA cert
 	if ca.File != nil {
 		err = ca.writeCert(cert)
 	}
 	// If there were no errors, update our internal notion of what the CA is.
 	if err != nil {
-		// see CA.Load(); strip prefix/trailing whitespace to ensure our bytes.Equal() checks don't false positive.
-		ca.pem = []byte(strings.TrimSpace(string(cert[:])))
+		ca.pem = cert
 	}
-
 	return true, err
 }
 
+// CompareCertificates x509 compares two CA certificates
+func CompareCertificates(cert1, cert2 []byte) (bool, error) {
+	p1, _ := pem.Decode(cert1)
+	if p1 == nil {
+		return false, errors.New("Unable to pem decode certificate")
+	}
+	parsedCert1, err := x509.ParseCertificate(p1.Bytes)
+	if err != nil {
+		return false, err
+	}
+	p2, _ := pem.Decode(cert2)
+	if p2 == nil {
+		return false, errors.New("Unable to pem decode certificate")
+	}
+	parsedCert2, err := x509.ParseCertificate(p2.Bytes)
+	if err != nil {
+		return false, err
+	}
+	return (parsedCert1.Equal(parsedCert2)), nil
```

Comment:
nit: the parens are unnecessary
```suggestion
	return parsedCert1.Equal(parsedCert2), nil
```